packages/expo-sqlite/ios/NativeDatabase.swift

```diff
@@ -2,7 +2,7 @@
 
 import ExpoModulesCore
 
-final class NativeDatabase: SharedRef<OpaquePointer?>, Equatable {
+final class NativeDatabase: SharedRef<OpaquePointer?>, Equatable, Hashable {
```

Comment:
What do you think if we make SharedObject already implement Equatable and Hashable? Seems like a common use case 🤔

Reply:
sqlite/next needs these because it has `cachedDatabases` and `cachedStatements`. i'm not clear whether that's a common case. typically people should keep the instances only in js. we could add the Equatable/Hashable support to SharedObject if there's another module need this.